[dynamo][numpy] Add support for builtin functions

larryliu0820 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #103457
* #97537

In order to be able to run stuff like:
```
def f(x):
	a = x.numpy()
        return a + a
```
This PR adds a branch in `BuiltinVariable` to handle `NumpyNdarrayVariable` case.

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @msaroufim

larryliu0820 committed:
[dynamo][numpy] Add support for builtin functions

In order to be able to run stuff like:
```
def f(x):
	a = x.numpy()
        return a + a
```
This PR adds a branch in `BuiltinVariable` to handle `NumpyNdarrayVariable` case.

[ghstack-poisoned]

ezyang commented:
Guessing this is not ready for review yet?

larryliu0820 commented:
Yeah let me fix the failures

larryliu0820 committed:
Update on "[dynamo][numpy] Add support for builtin functions"


In order to be able to run stuff like:
```
def f(x):
	a = x.numpy()
        return a + a
```
This PR adds a branch in `BuiltinVariable` to handle `NumpyNdarrayVariable` case.

cc voznesenskym penguinwu anijain2305 @EikanWang jgong5 @Guobing-Chen @XiaobingSuper zhuhaozhe blzheng @Xia-Weiwen wenzhe-nrv jiayisunx ipiszy

[ghstack-poisoned]

ezyang commented on torch/_dynamo/utils.py:
```diff
@@ -780,6 +780,21 @@ def check_unspec_python_args(args, kwargs):
+                for x in itertools.chain(args, kwargs.values())
+                if isinstance(x, NumpyNdarrayVariable)
+            ]
+        )
```
nit: you can do any test here

ezyang reviewed (approved):
Meh. It probably doesn't handle all the edges correctly but let's put it in for now.

larryliu0820 committed:
Update on "[dynamo][numpy] Add support for builtin functions"


In order to be able to run stuff like:
```
def f(x):
	a = x.numpy()
        return a + a
```
This PR adds a branch in `BuiltinVariable` to handle `NumpyNdarrayVariable` case.

cc voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy

[ghstack-poisoned]

larryliu0820 commented:
@pytorchbot merge

larryliu0820 commented:
@pytorchbot merge
